cpp/src/utilities/legacy/cuda_utils.hpp

```diff
@@ -79,8 +79,9 @@ struct scoped_stream {
     ~scoped_stream() {
         if (not std::uncaught_exception()) {
             auto synch_result = cudaStreamSynchronize(stream_);
+            assert(synch_result == cudaSuccess);
             if (synch_result == cudaSuccess) {
-                 cudaStreamDestroy(stream_);
+                 assert(cudaStreamDestroy(stream_) == cudaSuccess);
```

Comment:
This isn't what you want. In a release build, the `cudaStreamDestroy` won't be present because an `assert` is ifdef'd out.

Reply:
fixed this already.